### PR TITLE
Adding python-cmethods

### PR DESCRIPTION
**Pull request by btschwertfeger**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by btschwertfeger** [comment]
I am willing to be a maintainer for this package.

**Comment by btschwertfeger** [comment]
@conda-forge/help-python, ready for review!

File: recipes/python-cmethods/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  entry_points:
+    - "cmethods = cmethods:cli"
+
+requirements:
+  build:
+    - python >=3.8
+    - setuptools_scm
```
Comment (synapticarbors) [inline recipes/python-cmethods/meta.yaml:22]: `setuptools`, `setuptools_scm` and `wheel` should all be under `host` requirements, and then you can get rid of the `requirements`/`build` section. It's typically only needed if you are compiling code. See:
https://conda-forge.org/docs/maintainer/adding_pkgs/#host
https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#host

File: recipes/python-cmethods/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    and efficiency, python-cmethods offers a comprehensive suite of functions
+    tailored for applying bias correction methods to climate model simulations
+    and observational datasets via command-line interface and API.
+
+    Please cite this project as described in
+    https://zenodo.org/doi/10.5281/zenodo.7652755.
+  license: GPL-3.0-or-later
+  license_file: LICENSE
+  doc_url: https://python-cmethods.readthedocs.io/en/stable/
+  dev_url: https://python-cmethods.readthedocs.io/en/latest/
```
Comment (synapticarbors) [inline recipes/python-cmethods/meta.yaml:59]: The `dev_url` is typically the github repo, so here it should be the same as the `home` url.

File: recipes/python-cmethods/meta.yaml (hunk 1 above)
Comment (btschwertfeger) [reply recipes/python-cmethods/meta.yaml:23]: Thank you for sharing the relevant documentation!

**Pull request merged**